packages/x-components/tests/e2e/cucumber/mocked-responses.spec.ts

```diff
@@ -0,0 +1,558 @@
+            ...createFilter('Construye', false)
+          ])
+        ])
+      ],
+      results: [
+        createResultStub('LEGO Super Mario Pack Inicial: Aventuras con Mario - 71360', {
+          images: ['https://picsum.photos/seed/1/100/100'],
+          price: {
+            hasDiscount: false,
+            originalValue: 59.99,
+            value: 59.99
+          }
+        }),
+        createResultStub('LEGO Duplo Classic Caja de Ladrillos - 1091', {
+          images: ['https://picsum.photos/seed/2/100/100'],
+          price: {
+            hasDiscount: false,
+            originalValue: 29.99,
+            value: 29.99
+          }
+        }),
+        createResultStub('LEGO City Coche Patrulla de Policía - 60239', {
+          images: ['https://picsum.photos/seed/3/100/100'],
+          price: {
+            hasDiscount: false,
+            originalValue: 11.99,
+            value: 11.99
+          }
+        }),
+        createResultStub('LEGO City Police Caja de Ladrillos - 60270', {
+          images: ['https://picsum.photos/seed/4/100/100'],
+          price: {
+            hasDiscount: false,
+            originalValue: 39.99,
+            value: 39.99
+          }
+        }),
+        createResultStub('LEGO Friends Parque para Cachorros - 41396', {
+          images: ['https://picsum.photos/seed/5/100/100'],
+          price: {
+            hasDiscount: false,
+            originalValue: 11.99,
+            value: 11.99
+          }
+        }),
+        createResultStub('LEGO Creator Ciberdrón - 31111', {
+          images: ['https://picsum.photos/seed/6/100/100'],
+          price: {
+            hasDiscount: false,
+            originalValue: 11.99,
+            value: 11.99
+          }
+        }),
+        createResultStub('LEGO Technic Dragster - 42103', {
+          images: ['https://picsum.photos/seed/7/100/100'],
+          price: {
+            hasDiscount: false,
+            originalValue: 22.99,
+            value: 22.99
+          }
+        })
+      ],
+      totalResults: 7
+    });
+  }).as('interceptedResults');
+});
+
+Given('a second results API with a known response', () => {
+  cy.intercept('https://api.empathy.co/search', req => {
+    req.reply(<SearchResponse>{
+      banners: [],
+      promoteds: [],
+      facets: [],
+      results: [
+        createResultStub('LEGO Duplo Disney Tren de Cumpleaños de Mickey y Minnie - 10941', {
+          images: ['https://picsum.photos/seed/8/100/100']
+        }),
+        createResultStub('LEGO Disney Granja de Mickey Mouse y el Pato Donald - 10775', {
+          images: ['https://picsum.photos/seed/10/100/100']
+        })
+      ],
+      redirections: [],
+      partialResults: [],
+      totalResults: 7,
+      queryTagging: {
+        url: 'https://tagging.empathy.co',
+        params: {}
+      },
+      spellcheck: ''
+    });
+  }).as('interceptedNewResults');
+});
+
+Given('a results API with {int} results', (resultsLength: number) => {
```

Comment:
This is only used in the scroll feature, because of that you can remove the facets from the response and simplify it.